packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -703,29 +715,25 @@ export class GarbageCollector implements IGarbageCollector {
 			this.runtime.updateTombstonedRoutes(this.tombstones);
 		}
 
-		if (!this.configs.shouldRunSweep) {
-			return [];
-		}
-
-		// 1. Call the runtime to delete sweep ready nodes. The runtime returns a list of nodes it deleted.
-		// TODO: GC:Validation - validate that removed routes are not double delete and that the child routes of
-		// removed routes are deleted as well.
-		const deletedNodeIds = this.runtime.deleteSweepReadyNodes(Array.from(sweepReadyNodes));
-
-		// 2. Clear unreferenced state tracking for deleted nodes.
-		for (const nodeId of deletedNodeIds) {
-			const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
-			// TODO: GC:Validation - assert that the nodeStateTracker is defined
-			if (nodeStateTracker !== undefined) {
-				// Stop tracking so as to clear out any running timers.
-				nodeStateTracker.stopTracking();
-				// Delete the node as we don't need to track it any more.
-				this.unreferencedNodesState.delete(nodeId);
-			}
-			// TODO: GC:Validation - assert that the deleted node is not a duplicate
-			this.deletedNodes.add(nodeId);
+		if (this.configs.shouldRunSweep && sweepReadyNodes.size > 0) {
+			const sweepReadyDSAndBlobs = Array.from(sweepReadyNodes).filter((nodeId) => {
```

Comment:
I'd put a comment here or above `const contents` line explaining that we're dropping DDS IDs as an optimization since GC operates at DataStore granularity (like the one you have in `deleteSweepReadyNodes`)